examples/plot_gait.py

```diff
@@ -0,0 +1,225 @@
+"""
+Gait (steps) example
+====================
+
+In this example, we use DiCoDiLe on an open `dataset`_ of gait (steps)
+IMU time-series to discover patterns in the data.
+We will then use those to attempt to detect steps and compare our findings
+with the ground truth.
+
+.. _dataset: https://github.com/deepcharles/gait-data
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from dicodile.data.gait import get_gait_data
+from dicodile.utils.dictionary import init_dictionary
+from dicodile.utils.viz import display_dictionaries
+from dicodile.utils.csc import reconstruct
+from dicodile import dicodile
+
+###############################################################################
+# Retrieve trial data
+# -------------------
+
+trial = get_gait_data(subject=6, trial=1)
+
+###############################################################################
+# Let's have a look at the data for one trial.
+
+trial.keys()
+
+###############################################################################
+# We get a dictionary whose keys are metadata items, plus a 'data' key that
+# contains a numpy array with the trial time series for each sensor axis,
+# at 100 Hz resolution.
+
+# right foot acceleration (vertical)
+plt.plot(trial['data']['RAV'])
+
+###############################################################################
+# Let's look at a small portion of the series for both feet,
+# overlaid on the same plot
+
+fig, ax = plt.subplots()
+ax.plot(trial['data']['LAV'][5000:5800],
+        label='left foot vertical acceleration')
+ax.plot(trial['data']['RAV'][5000:5800],
+        label='right foot vertical acceleration')
+ax.set_xlabel('time (x10ms)')
+ax.set_ylabel('acceleration ($m.s^{-2}$)')
+ax.legend()
+
+###############################################################################
+# We can see the alternating left and right foot movements.
+#
+# In the rest of this example, we will only use the right foot
+# vertical acceleration.
+
+###############################################################################
+# Convolutional Dictionary Learning
+# ---------------------------------
+#
+# Now, let's use DiCoDiLe to learn patterns from the data and reconstruct
+# the signal from a sparse representation.
+#
+# First, we initialize a dictionary from parts of the signal:
+
+X = trial['data']['RAV'].to_numpy()
+X = X.reshape(1, *X.shape)
+
+print(X.shape)
+
+D_init = init_dictionary(X, n_atoms=8, atom_support=(300,), random_state=60)
```

Comment:
I just realized that the initialization of the dictionary did not have the `window` argument (it is not implemented yet).
Could you add this?

Also, could you try to run the example with `atom_support=175` or `250` to have more focused atoms?